packages/react-components/react-avatar/src/utils/partitionAvatarGroupItems.ts

```diff
@@ -0,0 +1,36 @@
+export type PartitionAvatarGroupItemsOptions<T> = {
+  items: readonly T[];
+  layout?: 'spread' | 'stack' | 'pie';
+  maxInlineItems?: number;
+};
+
+/**
+ * Get the inline items and overflowing items based on the array of AvatarGroupItems needed for AvatarGroup.
+ *
+ * @param options - Configure the partition options
+ *
+ * @returns Two arrays of AvatarGroupItems splitted between the inline items and the overflowing items
+ * based on maxItems.
+ *
+ * NOTE: Each item in the array must be a single AvatarGroupItem. If an item in the array contains a fragment with
+ * two avatars, the items will not be partitioned correctly.
+ */
+export const partitionAvatarGroupItems = <T>(options: PartitionAvatarGroupItemsOptions<T>) => {
+  const { items } = options;
+  const isPie = options.layout === 'pie';
+  const maxInlineItems = isPie ? 3 : options.maxInlineItems ?? 5;
+  const inlineCount = -(maxInlineItems - (items.length > maxInlineItems ? 1 : 0));
```

Comment:
Just a soft suggestion here -- I think I'd suggest splitting all this logic into an `if (isPie)`/not `isPie`, and also maybe separate out `hasOverflow` vs. not. Maybe something like this?

```js
const { items, maxInlineItems = 5 } = options;
const isPie = options.layout === 'pie';

if (isPie) {
  return {
    inlineItems: items.slice(0, 3),
    overflowItems: items,
  };
}

const hasOverflow = items.length > maxInlineItems;
const overflowIndex = -1 * (maxInlineItems - 1);

const inlineItems = hasOverflow ? items.slice(overflowIndex) : items;
const overflowItems = hasOverflow ? items.slice(0, overflowIndex) : [];

return {
  inlineItems,
  overflowItems,
};
```

(there's nothing wrong with your current approach -- I just found the separation of pie vs. not easier to read, since pie doesn't seem to share any splitting logic with the other layouts)

Reply:
that's great, I'll add that change. Better readability is always welcomed :)